### ISSUE TITLE
authentication breaks after wsus updates to 2012r2

### ISSUE THREAD

**Issue by hh** [criteo-cookbooks/wsus-client#18]
We need updates applied, and they can only be applied successfully via schtask criteo-cookbooks/wsus-client#16. And we need to do it from the base secure AMIs from CIS https://aws.amazon.com/marketplace/pp/B00UVT62LG

This spike sorta works, but I start getting 401 authentication fails after the wsus-updates apply.

At some point one of the updates seems to be wreaking havoc with auth... neither winrb nor rdp work afterwards. I'm trying to figure out at way to debug now... takes nearly 2 hours for updates to install.

``` ruby
require 'chef/provisioning/aws_driver'
require 'pry-byebug'
current_dir = File.dirname(__FILE__) # to get relative files... encrypted databag

with_chef_server "https://api.chef.io/organizations/#{ENV['CHEF_ORG']}",
                 :client_name => Chef::Config[:node_name],
                 :signing_key_filename => Chef::Config[:client_key]

with_machine_options bootstrap_options: {
                       instance_type: 'c3.2xlarge'
                     },
                     image_id: 'ami-7bc3e04b' # aws-marketplace/CIS Microsoft Windows Server 2012 R2 Benchmark v1.1.0-26bb465c-ce26-4da9-afb8-040b2f8c9a7f-ami-7a88f312.2

machine_name = 'base-2012-hardened-xv'

m = machine "#{machine_name}" do
  action :allocate
end

# converge a machine with chef-client::task
machine "#{machine_name}" do
  #action :nothing
  action :converge
  attribute %w{ client_client task frequency}, 'once'
  recipe 'chef-client::task'
  # these attributes will be use in the wsus convergence via schtask
  attribute %w{ wsus_client no_reboot_with_logged_users }, false
  attribute %w{ wsus_client automatic_update_behavior }, :install
  attribute %w{ wsus_client reboot_warning }, 1
  attribute %w{ wsus_client reboot_prompt_timeout }, 1
end

ruby_block "Run wsus-update via schtask on #{machine_name}" do
  #action :nothing
  block do
    mr=resources(machine: machine_name).provider_for_action(:ready)
    mr.load_current_resource
    machine=mr.action_ready
    driver = node.run_state[:chef_provisioning].drivers.values.first
    i=driver.ec2.instances[machine.machine_spec.reference['instance_id']]
    #not sure how to add recipes in a ruby_block...
    # it also seems that any machine 'thismachine' { recipe 'foo' }
    # resources that come later in __FILE__ update the run_list before
    # they execute, probably something to do with execution contexts
    ah = Chef::Provisioning::ChefProviderActionHandler.new(mr)
    if machine.execute_always(
         "knife node show #{machine_name} -a run_list -c c:\\chef\\client.rb"
       ).stdout.lines.grep(/wsus-client/).empty?
      machine.execute(ah,
        "knife node run_list add #{machine_name} wsus-client -c c:\\chef\\client.rb"
      )
    end
    machine.execute(ah, 'schtasks /run /tn chef-client')
    finished_running = false
    timer = 0
    until finished_running
      task_query = machine.execute_always('schtasks /query /tn chef-client').stdout.lines.grep(/chef-client/)
      finished_running = task_query.grep(/Running/).empty?
      Chef::Log.warn "chef-client wsus-client schtask running for #{timer} seconds"
      sleep 10
      timer += 10
    end
  end
end
```

**Comment by hh** [comment]
I realize there are a lot of assumptions in this ticket, but I have't found a public place to document this issue (it may be an issue with the CIS AMI, or with a particular windows update, it's not likely an issue specific to our cookbook, but whatever the resulting workaround, it will be useful to include it in the wsus-client cookbook)

**Comment by Annih** [comment]
Hello @hh 

With any chance did you succeed in your investigation? I'm not able to reproduce your issue with virtual box VM or bare metal servers :(

Regards,

Annih

**Comment by hh** [comment]
I had to open a support ticket with Amazon to track the issue down.
There is a windows update applied at some point (I didn't track down which one) that breaks early versions of Ec2Config service that ships with CIS images. Amazon said it's a known issue and I think they mean there is a reference in the release notes at https://aws.amazon.com/developertools/5562082477397515

I had to add a cookbook before the wsus upgrade that installs and restarts the latest Ec2Config service.
